content/00.zksync-era/40.tooling/20.hardhat/30.plugins/120.hardhat-zksync-node.md

```diff
@@ -74,120 +170,78 @@ npm run hardhat node-zksync
-- `--fork-block-number` - Specifies the block height at which to fork.
-- `--replay-tx` - Transaction hash to replay.
+### Testing with hardhat-zksync-node
 
-::callout{icon="i-heroicons-exclamation-triangle" color="amber"}
-**Parameter Restrictions**:
-The `--replay-tx` and `--fork-block-number` parameters cannot be specified simultaneously.
-The `--replay-tx` is used for replaying a remote transaction locally for deep debugging,
-while `--fork-block-number` is used for forking the blockchain at a specified block number.
-Combining these actions is not supported.
+The plugin enhances Hardhat's test task to run against anvil-zksync node:
 
-Additionally, if either `--replay-tx` or `--fork-block-number` is specified, the `--fork` parameter must also be provided.
-::
+1. Use the `hardhat` network with `zksync: true`
+1. Node starts automatically with tests
+1. Node shuts down after test completion
+1. Port allocation starts from 8011 to 8021 depends on availability
 
-::callout{icon="i-heroicons-information-circle" color="blue"}
-**Node task override**:
-If the network `zksync` flag is set to `true`, the `hardhat node` command will launch the `anvil-zksync` node on port `8545`.
-::
+## Troubleshooting
 
-::callout{icon="i-heroicons-information-circle" color="blue"}
-**Overrides for Deploy and Run tasks**:
-If the Hardhat network is selected and the `zksync` flag is set to true,
-the plugin will override the `hardhat run` command and all deployment tasks from the `hardhat-zksync-deploy` plugin,
-launching the `anvil-zksync` node before executing the scripts.
-
-**Note**:
-To achieve this for deployment scripts,
-the `hardhat-zksync-node` plugin must be imported after the `hardhat-zksync-deploy` plugin in the hardhat configuration file.
-::
+### Common Issues
 
-::callout{icon="i-heroicons-information-circle" color="blue"}
-**Learn More**:
-If you wish to learn more about replaying transactions or forking,
-check out the [anvil-zksync documentation](/build/test-and-debug/in-memory-node).
-::
+1. **Configuration Issues**
+   - Make sure `hardhat-zksync-node` is imported after `hardhat-zksync-deploy` in your config
+   - Verify network configuration has `zksync: true` set
+   - Check if anvil-zksync version is compatible with your setup
 
-::callout{icon="i-heroicons-information-circle" color="blue"}
-**Supported APIs**:
+2. **Provider Connection Issues**
+   - Ensure the node is running before executing commands
+   - Verify RPC URL is correct and accessible
+   - Check if you have sufficient permissions to access the node
 
-To see a list of all supported APIs, visit [this link](%%zk_git_repo_era-test-node%%/blob/main/SUPPORTED_APIS.md).
-::
+3. **Cache and Storage Issues**
+   - Clear cache using `--reset-cache` flag if experiencing unexpected behavior
+   - Check disk space for cache directory
+   - Verify write permissions in cache directory
 
-## Running Hardhat's test Task with hardhat-zksync-node
+### Specific Error Messages
```

Comment:
I think we can remove this section and keep it all under trouble shooting but update format to keep consistent with other troubleshooting sections (e.g. Problem / Solution)